certora/specs/SocialRecoveryModule.spec

```diff
@@ -436,3 +436,25 @@ rule canFinalizeRecoveryOnlyAfterDelayPeriod(env e) {
 
     assert success => require_uint64(e.block.timestamp) >= recoveryTimestamp, "Recovery finalized before delay period";
 }
+
+// Recovery can be finalized by anyone after the delay period is over.
+rule anyoneCanFinalizeRecovery(env e) {
```

Comment:
This rule here https://github.com/5afe/CandideWalletContracts/blob/6bc526219a4c66e2df61b67ae0dcba66b938c2e2/certora/specs/SocialRecoveryModule.spec#L428 already verifies that any address can finalize recovery. How is this rule different from one already implemented?

Reply:
That rule only verifies if the transaction went through, then the delay period should be complete. This one checks both conditions (success & failure to the `finalizeRecovery` based on `delayPeriod`). Though I do agree both have some similarities.

Reply:
But, in both the rules any address can finalise the recovery though assertion conditions differ.

Reply:
Yes, hence the similarity. This one is a bit more open in terms of assertion but a bit more stricter in terms of require. The other one is the opposite.

Reply:
Honestly I'm not getting why this rule is needed. This rule verifiers essentially the same thing, but with the more stricter assertion, and this stricter assertion requires a more polished pre-state to prevent other revert paths from happening. I don't see a bug this rule catches that the other one don't

Reply:
agreed, I think the same.

Reply:
So, I talked with @mmv08 and thought of merging both the rules to check for most cases. Updated the rule based on that.

Reply:
I was in favour of merging this and removing the other one by the way, since this one already checks what the other one checks